frappe/custom/doctype/doctype_layout/doctype_layout.json

```diff
@@ -1,6 +1,6 @@
 {
  "actions": [],
- "allow_rename": 1,
+ "allow_rename": 0,
```

Comment:
??

Reply:
There is no point in allowing renaming. There should ideally only be 1 DocType Layout for a specific DocType.

Reply:
How not allowing rename will solve that?
I can just go and create another Doctype Layout for the same Doctype with a different name.

If I create a layout for doctype with a specific name later if I want to change it I will have to create another layout instead we should allow renaming.

We can have multiple layouts for the same doctype (I don't see any problem in this)

Reply:
> How not allowing rename will solve that? I can just go and create another Doctype Layout for the same Doctype with a different name.

With naming set to `"autoname": "field:document_type"`, you basically can't create multiple ones for the same DocType.

> We can have multiple layouts for the same doctype (I don't see any problem in this)

My understanding of DocType Layout seems to be wrong(lack of documentation and too lazy to check the code). Could you explain how it would work in case of multiple layouts?

Reply:
We can create multiple Doctype Layout for one Doctype. E.g.
For ToDo in one layout, only a few fields will be visible and in another layout, some other fields will be visible, but for both routes will be different.

For E.g

route: task-todo
fields visible: task id, task name etc

route: task-project
fields visible: project id, project name etc


And if you will use document_type as autoname then we will not be able to create unique route which will not work.